tutorials/tutorial14_bottlenecks.ipynb

```diff
@@ -0,0 +1,390 @@
+    "This tutorial walks you through the process of *running the bottlenecks experiments*. The bottleneck experiment, depicted in Fig. 1, is intended to illustrate the dynamics of traffic in a bottleneck. In particular, our bottleneck is intended to imitate the dynamics of traffic on the San Francisco-Oakland Bay Bridge, where fifteen (?) lanes narrow to eight (?) to five (?). In our bottleneck, we have N * 4 lanes narrow to N * 2 to N, where N is a scaling factor that can be used to increase the number of lanes. As demonstrated in Fig. 2, this bottleneck has a phenomenon known as *capacity drop*: as the number of vehicles flowing into the system increases the number of exiting vehicles initially increases. However, above a critical flow of entering vehicles, the output of the system starts to decrease as congestion forms. Just like in the San Francisco-Oakland Bay Bridge, there is a configurable toll booth and a set of traffic lights that can be used to control the inflow. Each of the merges is implemented as a zipper merge, in which the vehicles merge as late as possible. \n",
+    "\n",
+    "<center>\n",
+    "<img src=\"img/bottleneck.png\">\n",
+    "\n",
+    "Figure 1. A visual of the bottleneck that is modeled on the San Francisco-Oakland bay bridge. There are two bottlenecks, the first cuts the number of lanes in half and the second cuts the number of lanes in half again.</center>\n",
+    "\n",
+    "<center>\n",
+    "<img src=\"img/capacity_curve.png\">\n",
+    "</center>\n",
+    "<center>\n",
+    "Figure 2. The inflow-outflow relationship for the bottleneck depicted in Fig. 1. As the inflow increases, the outflow steadily ticks up, but above a critical inflow the outflow suddenly ticks down and remains at that value. This phenomenon is known as *capacity drop*. The shaded area indicates 1 std deviation around the outflow.\n",
+    "</center>\n",
+    "\n",
+    "\n",
+    "The remainder of this tutorial is organized as follows:\n",
+    "\n",
+    "* Section 1 introduces the basic configurable parameters of the traffic light environment.\n",
+    "* Section 2 shows how to configure the toll booth.\n",
+    "* Section 3 shows how to use the traffic lights to perform ramp metering.\n",
+    "* Section 4 introduces some autonomous vehicles into the system and describes the Markov Decision Process that the bottleneck comes pre-configured with. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Configuring the Environment\n",
+    "\n",
+    "Here we describe the different basic configurable parameters of the bottleneck environment. First, we import the necessary classes and parameters to run the environment. We will highlight the effects of changing certain parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from flow.core.params import SumoParams, EnvParams, NetParams, InitialConfig, \\\n",
```

Comment:
should we document this code block for thoroughly? i.e. write in front of every few lines in comments what's going on